main/sceneBuilder.cpp

```diff
@@ -0,0 +1,75 @@
+#include "gtest/gtest.h" // https://google.github.io/styleguide/cppguide.html#Namespaces
+#include "main/sceneBuilder.h"
+#include "obj/generators.h"
+#include "obj/objects.h"
+
+using core::Ray;
+using core::RayHitData;
+using core::Vec3;
+using generators::PointSource;
+using objects::Object;
+using objects::TriangleObj;
+
+const double kSkipFreq = 1000;
+generators::FakeRandomGen fakeRand;
+
+TEST(SceneBuildertest, IsTestSampleRightShape)
+{
+    Vec3 k1(-0.5, -0.5, 0);
+    Vec3 k2(0.5, -0.5, 0);
+    Vec3 k3(-0.5, 0.5, 0);
+    Vec3 k4(0.5, 0.5, 0);
+    Vec3 k5(123, 123, 1);
+
+    SceneBuilder builder;
+
+    const std::vector<TriangleObj> referenceSample = {TriangleObj(k1, k2, k3), TriangleObj(k2, k3, k4)};
+
+    // because how operator== in triangleObj is implemented, I can do that:
+
+    std::vector<std::unique_ptr<Object>> testSample = builder.BuildTestSample();
+    ASSERT_TRUE(std::find(referenceSample.cbegin(), referenceSample.cend(), *testSample[0]) != referenceSample.cend() &&
```

Comment:
Please make sure to check what GTest has to offer. You use ASSERT_TRUE way to much.

For example for containers there is something like:

```
ASSERT_THAT(test_sample, UnorderedElementsAre(TriangleObj(k1, k2, k3), TriangleObj(k2, k3, k4)));
```

This likely does not compile because of the unique_ptrs but you should get rid of them anyway I think (see the other comment).